### PR TITLE
Cleaned up getting-started docs, added more docs for Querier.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
@s-urbaniak Adding this section specially for you: https://deploy-preview-1541--thanos-io.netlify.com/components/query.md/#querier-use-cases-why-do-i-need-this-component 

Still in progress, but early feedback would be nice (:

File: docs/getting-started.md [hunk 1]
```diff
@@ -67,210 +72,42 @@ of the community. Here are ways to get in touch with the community:
...
-```bash
-thanos sidecar \
-    --tsdb.path                 /var/prometheus \
-    --objstore.config-file      bucket_config.yaml \       # Bucket config file to send data to
-    --prometheus.url            http://localhost:9090 \    # Location of the Prometheus HTTP server
-    --http-address              0.0.0.0:19191 \            # HTTP endpoint for collecting metrics on the Sidecar
-    --grpc-address              0.0.0.0:19090              # GRPC endpoint for StoreAPI
-```
+* [prometheus-operator](https://github.com/coreos/prometheus-operator): Prometheus operator has support for deploying Prometheus with Thanos
+* [kube-thanos](https://github.com/metalmatze/kube-thanos): Jsonnet based Kubernetes templates.
```
Comment (kakkoyun) [inline docs/getting-started.md]:
```suggestion
* [kube-thanos](https://github.com/thanos-io/kube-thanos): Jsonnet based Kubernetes templates.
```

File: docs/components/query.md [hunk 2]
```diff
@@ -4,23 +4,67 @@ type: docs
...
+
+* Prometheus (see [Sidecar](sidecar.md))
+* Object Storage (see [Store Gateway](store.md))
+* Global alerting/recording rules evaluations (see [Ruler](rule.md))
+* Handled metrics from the remote write streams (see [Thanos Receiver](../proposals/201812_thanos-remote-receive.md))
+* Another Querier (you can stack Queriers on top of each other)
+* Non-Prometheus systems!
+    * e.g [OpenTSDB](../integrations.md#opentsdb)
+
+Thanks to that, you can run queries (manually, from Grafana or Alert rule) that aggregate metrics from mix of those sources.
```
Comment (povilasv) [inline docs/components/query.md]:
```suggestion
Thanks to that, you can run queries (manually, from Grafana or via Alerting rule) that aggregate metrics from mix of those sources.
```

File: docs/components/query.md [hunk 3]
```diff
@@ -4,23 +4,67 @@ type: docs
...
+
+* `sum(cpu_used{cluster=~"cluster-(eu1|eu2|eu3|us1|us2|us3)", job="service1"})` that will give you sum of CPU used inside all listed clusters for service `service1`. This will work
+even if those clusters runs multiple Prometheus servers each. Querier will know which data sources to query.
+
+* In single cluster you shard Prometheus functionally or have different Prometheus instances for different tenants. You can spin up Querier to have access to both within single Query evaluation.
+
+### Run-time deduplication of HA groups
 
-## Deduplication
+Prometheus is stateful and does not allow to replicate its database. This means that increasing high availability by just running multiple replicas of Prometheus is not very easy to use.
```
Comment (povilasv) [inline docs/components/query.md]:
```suggestion
Prometheus is stateful and does not allow replicating its database. This means that increasing high availability by running multiple Prometheus replicas is not very easy to use.
```

File: docs/quick-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+Always make sure to run Prometheus as recommended by Prometheus team, so:
+
+* Put Prometheus in the same failure domain. This means same network, same datacenter as monitoring services.
+* Use persistent disk to persist data across Prometheus restarts.
+* Use local compaction for longer retentions.
+* Do not change min TSDB block durations. 
+* Do not scale out Prometheus unless necessary. Single Prometheus is highly efficient (: 
+
+However if you scale out your Prometheus instance, Thanos is exactly helping with that.
```
Comment (povilasv) [inline docs/quick-tutorial.md]:
```suggestion
We recommend using Thanos when you need to scale out your Prometheus instance.
```

File: docs/quick-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+* Do not change min TSDB block durations. 
+* Do not scale out Prometheus unless necessary. Single Prometheus is highly efficient (: 
+
+However if you scale out your Prometheus instance, Thanos is exactly helping with that. 
+
+### Components
+
+Following the [KISS](https://en.wikipedia.org/wiki/KISS_principle) and Unix philosophies, Thanos is made of a set of components with each filling a specific role.
+
+* Sidecar: connects to Prometheus and reads its data for query and/or upload it to cloud storage.
```
Comment (povilasv) [inline docs/quick-tutorial.md]:
```suggestion
* Sidecar: connects to Prometheus, reads its data for query and/or uploads it to cloud storage.
```

File: docs/quick-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+* Store Gateway: exposes the content of a cloud storage bucket.
```
Comment (povilasv) [inline docs/quick-tutorial.md]:
```suggestion
* Store Gateway: serves metrics inside of a cloud storage bucket.
```

File: docs/quick-tutorial.md [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+* Compactor: compacts, downsamples and applies retention on the data stored in remote storage.
```
Comment (povilasv) [inline docs/quick-tutorial.md]:
```suggestion
* Compactor: compacts, downsamples and applies retention on the data stored in cloud storage bucket.
```

File: docs/quick-tutorial.md [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+
+See those components on this diagram:
+
+<img src="img/arch.jpg" class="img-fluid" alt="architecture overview" />
+
+### [Sidecar](components/sidecar.md)
+
+Thanos integrates with existing Prometheus servers through a [Sidecar process](https://docs.microsoft.com/en-us/azure/architecture/patterns/sidecar#solution), which runs on the same machine or in the same pod as the Prometheus server. 
+
+The purpose of the Sidecar is to backup Prometheus data into an Object Storage bucket, and giving other Thanos components access to the Prometheus instance the Sidecar is attached to via a gRPC API.
```
Comment (povilasv) [inline docs/quick-tutorial.md]:
```suggestion
The purpose of the Sidecar is to backup Prometheus data into an Object Storage bucket, and give other Thanos components access to the Prometheus metrics via a gRPC API.
```

**Pull request merged**